funnel/models/space.py

```diff
@@ -329,6 +329,16 @@ def fetch_sorted(cls):
         # union_all() because union() doesn't respect the orders mentioned in subqueries
         return upcoming.union_all(past)
 
+    def roles_for(self, actor=None, anchors=()):
+        roles = super(ProposalSpace, self).roles_for(actor, anchors)
+        if actor is not None:
+            if self.admin_team in actor.teams:
+                roles.add('admin')
+            if self.review_team in actor.teams:
+                roles.add('reviewer')
+        roles.update(self.profile.roles_for(actor, anchors))
+        return roles
```

Comment:
Please add a `reader` role here (available to all; [see notes](http://coaster.readthedocs.io/en/latest/sqlalchemy/roles.html)). In future we'll have private events where you must explicitly be granted a `reader` role.

Reply:
Also, in `Proposal.roles_for`, explicitly revoke the `reader` role if the state is DRAFT.